features/facebookImport/src/components/advertisingValueMiniStory/advertisingValueMiniStory.jsx

```diff
@@ -1,9 +1,33 @@
-import React from "react";
+import React, { useRef } from "react";
 import i18n from "../../i18n";
 
 import "./advertisingValueMiniStory.css";
 
+const calculateFontSize = (text, maxWidth) => {
+    // TODO: Extract text size affecting styles from target element
+    const minFontSize = 14;
+    const maxFontSize = 34;
+    const canvas = document.createElement("canvas");
+    const context = canvas.getContext("2d");
+    for (let fontSize = maxFontSize; fontSize > minFontSize; fontSize--) {
+        context.font = `${fontSize}px Jost`;
+        if (context.measureText(text).width <= maxWidth) return fontSize;
+    }
+    return minFontSize;
+};
+
 const AdvertisingValueMiniStory = ({ randomAdInterests, numberInterests }) => {
+    // following takes the three interests, splits them into single words and picks the longest word
+    let splitInterests = [].concat.apply(
+        [],
+        randomAdInterests.map((int) => int.split(" "))
+    );
+    let interestsLength = splitInterests.map((word) => word.length);
+    let longestInterestWordIndex = interestsLength.indexOf(
+        Math.max(...interestsLength)
+    );
+    let longestInterestWord = splitInterests[longestInterestWordIndex];
```

Comment:
OK, this is more efficient than what was in here before, thanks for the change. However, both have a problem. Imagine the longest interests are "iiiii" and "mmmmm". This is going to return the first word with that length, it could be any of them. Imagine it's "iiiii". Since what you're interested in is the print length, this could actually be a problem. Would it be possible to calculate the max print length here instead of the number of characters?

Reply:
Well, since you are mentioning. I didn't think about it long enough earlier, but  I guess I could use the context.measureText-logic from above. Or do you have a better idea?